### PR TITLE
satellite/repair: delete pieces that failed piece hashes verification from pointer

### PR DESCRIPTION
**Pull request by VinozzZ**
What: 

delete corrupted piece from pointer during repair.

Why:

If a piece failed piece hash verification, we should remove it from pointer

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: satellite/repair/repair_test.go [hunk 1]
```diff
@@ -706,3 +799,41 @@ func stopNodeByID(t *testing.T, ctx context.Context, planet *testplanet.Planet,
 		}
 	}
 }
+
+// nolint:golint
```
Comment (egonelbre) [inline satellite/repair/repair_test.go]: Remove this

File: satellite/repair/repairer/ec.go [hunk 2]
```diff
@@ -117,7 +117,10 @@ func (ec *ECRepairer) Get(ctx context.Context, limits []*pb.AddressedOrderLimit,
 				if err != nil {
 					// gather nodes that failed to match piece hash with the original piece hash
```
Comment (egonelbre) [inline satellite/repair/repairer/ec.go]: `// gather nodes that failed to match piece hash with the original piece hash` shouldn't this be `gather nodes where the calculated piece hash doesn't match the uplink signed piece hash`

File: satellite/repair/repair_test.go [hunk 3]
```diff
@@ -706,3 +799,41 @@ func stopNodeByID(t *testing.T, ctx context.Context, planet *testplanet.Planet,
 		}
 	}
 }
+
+// corruptPieceData manipulates piece data on a storagde node.
```
Comment (mobyvb) [inline satellite/repair/repair_test.go]: `storagde` -> `storage`

File: satellite/repair/repairer/ec.go [hunk 4]
```diff
@@ -115,9 +115,12 @@ func (ec *ECRepairer) Get(ctx context.Context, limits []*pb.AddressedOrderLimit,
...
 				inProgress--
 				if err != nil {
-					// gather nodes that failed to match piece hash with the original piece hash
+					// gather nodes where the calculated piece hash doesn't match the uplink signed piece hash
 					if ErrPieceHashVerifyFailed.Has(err) {
-						failedNodes = append(failedNodes, limit.GetLimit().StorageNodeId)
+						failedPieces = append(failedPieces, &pb.RemotePiece{
+							PieceNum: int32(currentLimitIndex),
+							NodeId:   limit.GetLimit().StorageNodeId,
+						})
```
Comment (ethanadams) [inline satellite/repair/repairer/ec.go]: `currentLimitIndex` seems to be the array index.  does the array index always equal the `PieceNum`?

Comment (mobyvb) [reply satellite/repair/repairer/ec.go]: yeah, the limits are generated with `orders.CreateGetRepairOrderLimits`, which uses the piece number to index the limits.

**Pull request merged**